### ISSUE TITLE
Update Prepared Statement - wrong order of parsing

### ISSUE THREAD

**Issue by mmatloka** [outworkers/phantom#455]
I have a query

```
update()
.p_where(_.someString eqs ?)
.p_modify(_.someDate setTo ?)
.p_and(_.sthElse setTo ?)
.prepare()
```

but I got error saying that there is no codec between string and timestamp. It seems that in `BoundStatement` order of codecs in `wrapper.codecFor(i, ` is different and it tries to parse `someString` as `someDate`.

I have seen a javadoc saying
```
*   database.table.column.update.p_modify(_.bla setTo ?).where(_.a eqs ?)
```
but this does not compile. But probably it is how it is interpreted underneath.

**Comment by alexflav23** [comment]
Hi @mmatloka,

I'm investigating this, it looks like automagically the Java Driver drops some of the arguments, I'm not sure where the culprit is currently. I'm going to need some more time on this one I'm afraid, but thank you for reporting it.

I think `BoundStatement` either doesn't really account for order or it's because the Phantom query builder can inverse the statements irrespective of the order of specification based on merge clauses that this happen.

I'll have a look at setting the modify clause first, see if that changes anything at all.

Regards.

**Comment by alexflav23** [comment]
Hi @mmatloka,

Fixed in `1.25.0`, going live today/tomorrow morning. Thank you once again for reporting it. The breaking change is that you will have to specify the modified arguments first when you bind to a prepared update query. This will force a match. Changeset is available in the PR, but essentially we build to separate argument type lists, one for where->and, and another for the `SET` part, then join them together at the end.

Regards.
